/// @file
/// @author Thaddeus Diamond <diamond@cs.yale.edu>
/// @version 0.1
/// @since 26 Jan 2012
///
/// @section DESCRIPTION
///
/// This is a TPC-C specific serializable argset

message TPCCArgs {
  // This represents the system time for the transaction
  optional double system_time = 1;

  // Number of warehouses in an initialize txn or load generation
  optional bool multipartition = 10;

  // This represents the total order line count for a new order and an array
  // of the quantity of item per supply warehouse
  optional int32 order_line_count = 20;
  repeated int32 quantities = 21;

  // In a payment transaction, this represents the amount of payment
  // and the last name and previous last name (which if
  // it does not match means the last name was recently looked up)
  optional int32 amount = 31;
  optional bytes last_name = 32;

  // In a delivery transaction, this represents O_CARRIER_ID and OL_DELIVERY_D
  optional int32 carrier_id = 41;
  optional int32 delivery_date = 42;

  // In a stock-level transaction, we keep track of the minimum stock threshold
  // we are trying to read out
  optional int32 threshold = 51;
}
